subsys/logging/log_core.c

```diff
@@ -282,7 +282,7 @@ void log_hexdump_sync(struct log_msg_ids src_level, const char *metadata,
 
 static u32_t timestamp_get(void)
```

Comment:
in that form it will not be optimal as on every timestamp_get there will be runtime evaluation of cycles per sec.
You can remove that function and instead do something like in log_core_init:
```
timestamp_func = (sys_clock_hw_cycles_per_sec() > 1000000) ? k_uptime_get_32 : k_cycle_get_32;
```
which replaces:
```
timestamp_func = timestamp_get;
```

Reply:
Updated. Please check the solution.